src/Shared/Buffers.MemoryPool/PinnedBlockMemoryPoolMetrics.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.Metrics;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+#nullable enable
+
+namespace System.Buffers;
+
+internal sealed class PinnedBlockMemoryPoolMetrics
+{
+    // Note: Dot separated instead of dash.
+    public const string MeterName = "System.Buffers.PinnedBlockMemoryPool";
+
+    private readonly Meter _meter;
+    private readonly UpDownCounter<long> _currentMemory;
+    private readonly Counter<long> _totalAllocatedMemory;
+    private readonly Counter<long> _evictedMemory;
+    private readonly Counter<long> _usageRate;
+
+    public PinnedBlockMemoryPoolMetrics(IMeterFactory meterFactory)
+    {
+        _meter = meterFactory.Create(MeterName);
```

Comment:
It looks like there could be multiple instances of memory pool used across assemblies with shared source. Fortunatly I believe that as long as the same `IMeterFactory` is used for all of them, the meter and counters created will be the same, and values will be saved to the same location. However, I think a unit test to double check that is important.

Reply:
Added a test and updated our TestMeterFactory implementation to return the same Meter instance for similar meters. Same as the default meter factory impl: https://source.dot.net/#Microsoft.Extensions.Diagnostics/Metrics/DefaultMeterFactory.cs,37